workflow/executor.py

```diff
@@ -320,7 +341,7 @@ def _load_resource(logger: logging.Logger, env_config: EnvConfig, rai_config: Ra
             if ContainerType.LOCAL == container.type or ContainerType.AZURE == container.type:
                 rai.execute_query(logger, rai_config, q.load_resources(logger, config, resources, src), readonly=False)
             elif ContainerType.SNOWFLAKE == container.type:
-                snow.sync_data(logger, config, rai_config, resources, src)
+                snow.begin_data_sync(logger, config, rai_config, resources, src)
```

Comment:
There can be parallel writes to the same DB with different engines: specified in CLI arguments and ingestion engines created by Snowflake integration service.